3rd_party/CodeCoverage.cmake

```diff
@@ -203,7 +203,7 @@ mark_as_advanced(
 
 get_property(GENERATOR_IS_MULTI_CONFIG GLOBAL PROPERTY GENERATOR_IS_MULTI_CONFIG)
 if(NOT (CMAKE_BUILD_TYPE STREQUAL "Debug" OR GENERATOR_IS_MULTI_CONFIG))
-    message(WARNING "Code coverage results with an optimised (non-Debug) build may be misleading")
+    # message(WARNING "Code coverage results with an optimised (non-Debug) build may be misleading")
```

Comment:
Is it really a good idea to rely on everyone knowing about this? Are there any downsides of leaving this warning here?

Reply:
I though about that, the more fundamental problem here is, that the warning is wrong from my point of view.  The CodeCoverage.cmake script itself adds "-g" already.  Using the debug build type might globaly define "DEBUG=true", which in turn also requires "all" dependent libraries to be build in debug mode (otherwise API incompatibilities might occur, because for example libstd++ was built in non-debug mode and you're linking against it while using <vector> which uses the global DEBUG flag which in turn makes it incompatible ...)
This is really tricky, and indeed, you really need to know what you're doing.  In our case I don't see any reason to build everything in DEBUG mode, so this warning will be always there - which I don't really like.
And as stated, it is just not true, you need to take care anyway of the the compilation flags and know what you're doing ...

Reply:
That reasoning totally makes sense